client/block.c

```diff
@@ -1935,6 +1920,14 @@ void remove_orphan(struct block_internal* bi, int remove_action)
 					for (i = 0; i < bi->nlinks; ++i) {
 						remove_orphan(bi->link[i], ORPHAN_REMOVE_NORMAL);
 					}
+					if(bi->flags & BI_REMARK){
```

Comment:
Remark for BI_EXTRA block will be added only if/when those blocks are not BI_EXTRA anymore.
This was a bug of the other remark implementation